src/pages/Address/index.tsx

```diff
@@ -8,13 +8,25 @@ import { AddressTransactions, AddressOverview } from './AddressComp'
 import { fetchAddressInfo, fetchTransactionsByAddress } from '../../service/http/fetcher'
 import { QueryResult } from '../../components/QueryResult'
 import { defaultAddressInfo } from './state'
-import { usePaginationParamsInPage } from '../../utils/hook'
+import { usePaginationParamsInPage, useSearchParams, useSortParam } from '../../utils/hook'
 import { isAxiosError } from '../../utils/error'
 
+export type TxTypeType = 'outgoing' | 'incoming' | 'customised' | undefined
+
+function isTxFilterType(s?: string): s is TxTypeType {
+  return s === 'outgoing' || s === 'incoming' || s === 'customised' || s === undefined
```

Comment:
array.includes can be used here

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes